source/nodejs/adaptivecards/src/utils.ts

```diff
@@ -25,21 +25,43 @@ export function appendChild(node: Node, child: Node) {
 }
 
 export function setProperty(target: any, propertyName: string, propertyValue: any, defaultValue: any = undefined) {
-	if (propertyValue && (!defaultValue || defaultValue !== propertyValue)) {
+    let targetIsAlreadySet = target[propertyName] !== undefined;
+
+    if (propertyValue === undefined || propertyValue === null) {
+        if (!targetIsAlreadySet) {
```

Comment:
shouldn't it be complement of this?
e.g. if (already set) then delete?

Reply:
No. This is for round-tripping; if the property is already set and the new value is null, we just leave the property as is.

Reply:
i see; but I felt weird that we are deleting it when it's not set.

---
In reply to: [276771114](https://github.com/Microsoft/AdaptiveCards/pull/2714#discussion_r276771114) [](ancestors = 276771114)